src/compiler/checker.ts

```diff
@@ -10664,6 +10731,12 @@ namespace ts {
             }
 
             forEach(node.members, checkSourceElement);
+
+            // Classes containing abstract methods must be marked abstract
+            if (!(node.flags & NodeFlags.Abstract) && forEach(node.members, element => element.flags & NodeFlags.Abstract)) {

```

Comment:
looking at this again, i would say this should be done when check methods, this way we do not have to visit all members of all classes one more time.